core/lib/state/src/rocksdb/mod.rs

```diff
@@ -131,23 +175,111 @@ impl RocksdbStorage {
         tracing::info!(
             "Secondary storage for L1 batch #{latest_l1_batch_number} initialized, size is {estimated_size}"
         );
+
+        self.save_missing_enum_indices(conn).await;
+    }
+
+    async fn save_missing_enum_indices(&self, conn: &mut StorageProcessor<'_>) {
+        if let (Some(start_from), true) = (
+            self.enum_migration_start_from(),
+            self.enum_index_migration_chunks > 0,
+        ) {
+            let started_at = Instant::now();
+            let chunk_from = u16::from_be_bytes(start_from);
+            let chunk_to = chunk_from.saturating_add(self.enum_index_migration_chunks - 1);
+            tracing::info!("RocksDB enum index migration is not finished, processing chunks {chunk_from}..={chunk_to}");
+
+            let mut write_batch = self.db.new_write_batch();
+            let mut number_of_keys_migrated = 0;
+            for chunk in chunk_from..=chunk_to {
+                let (keys, values): (Vec<_>, Vec<_>) = self
+                    .db
+                    .prefix_iterator_cf(StateKeeperColumnFamily::State, &chunk.to_be_bytes())
+                    .filter_map(|(key, value)| {
+                        (key.as_ref() != Self::BLOCK_NUMBER_KEY
+                            && key.as_ref() != Self::ENUM_INDEX_MIGRATION_PROGRESS_KEY
+                            && key.as_ref() != Self::ENUM_INDEX_MIGRATION_FINISHED)
+                            .then(|| (H256::from_slice(&key), H256::from_slice(&value[..32])))
+                    })
+                    .unzip();
+                let indices = conn
+                    .storage_logs_dedup_dal()
+                    .enum_indices_for_keys(&keys)
+                    .await;
+                number_of_keys_migrated += indices.len();
+                for ((key, value), index) in keys.into_iter().zip(values).zip(indices) {
+                    write_batch.put_cf(
+                        StateKeeperColumnFamily::State,
+                        key.as_bytes(),
+                        &Self::serialize_state_value(value, index),
+                    );
+                }
+            }
+
+            if chunk_to == u16::MAX {
```

Comment:
Why? Are we going to process only 65536 batches? (it's a default value for EN node now)

Reply:
Not sure what you mean by batch here. If L1 batches then no, if chunks then yes because 65536 chunks cover all keys. Anyway, approach will be changed according to https://github.com/matter-labs/zksync-era/pull/162#discussion_r1347307942